src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.Adapt.cs

```diff
@@ -0,0 +1,198 @@
+﻿// <copyright file="ColorSpaceConverter.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.ColorSpaces.Conversion
+{
+    using System;
+
+    using ImageSharp.ColorSpaces;
+    using ImageSharp.ColorSpaces.Conversion.Implementation.Rgb;
+
+    /// <content>
+    /// Performs chromatic adaptation on the various color spaces.
+    /// </content>
+    public partial class ColorSpaceConverter
```

Comment:
All cool. 👍  I've internalized all of it for now. 

The only reason `IColorConversion.Convert()` is not static is because some converters require the specific whitepoint to be passed. I guess we could refactor that to pass the whitepoint. 

`IChromaticAdaptation` exists as there are multiple possible adaption types, we only use `VonKriesChromaticAdaptation` for now though. However, there's no need for us to be passing `IColorConversion` etc through the constructor, they're concrete with no alternative methods.

Bear in mind it's already 96% faster than the original code from Colourful! 😄